### PR TITLE
refactor/perf: small perf. adjustments and fix logical nitpicks

### PR DESCRIPTION
**Pull request by collierrgbsitisfise**
- remove unused dep.
- perf. adjustments: combine independent async operation under `Promise.all`
- fix logical(not critical) nitpicks.(I'm not sure if they bring value into the project or not)

### CODE REVIEW

File: packages/extension/svelte-stuff/shared/io.ts [hunk 1]
```diff
@@ -5,10 +5,7 @@ export const getSocket = () => {
   if (!socket) {
     socket = new ReconnectingWebSocket(
       () =>
-        (apiBaseUrl.includes("https")
-          ? apiBaseUrl.replace("https", "wss")
-          : apiBaseUrl.replace("http", "ws")) +
-        `?accessToken=${accessToken}&refreshToken=${refreshToken}`
+        `${apiBaseUrl.replace("http", "ws")}?accessToken=${accessToken}&refreshToken=${refreshToken}`
```
Comment (RazerMoon) [inline packages/extension/svelte-stuff/shared/io.ts:8]: What if the base url includes "https" instead of "http"? Please re-check this line

**Review by RazerMoon — CHANGES_REQUESTED** [review]

File: packages/extension/svelte-stuff/shared/io.ts (hunk 1 above)
Comment (collierrgbsitisfise) [reply packages/extension/svelte-stuff/shared/io.ts:8]: @RazerMoon 
```javascript
 `https://some.host/api`.replace('http', 'ws'); // wss://some.host/api
```

Still should works fine, cause `https` and `wss` contain common last `s`.
Maybe I'm not correctly understand the question, let me know. Thanks !

Comment (RazerMoon) [reply packages/extension/svelte-stuff/shared/io.ts:8]: I didn't see that, you're right. My apologies 😅

**Review by benawad — CHANGES_REQUESTED** [review]

File: packages/extension/src/extension.ts [hunk 2]
```diff
@@ -28,12 +28,14 @@ export function activate(context: vscode.ExtensionContext) {
 
   context.subscriptions.push(
     vscode.commands.registerCommand("vsinder.reloadSidebar", async () => {
-      await vscode.commands.executeCommand("workbench.action.closeSidebar");
-      await vscode.commands.executeCommand(
-        "workbench.view.extension.vsinder-sidebar-view"
-      );
-      setTimeout(() => {
+      await Promise.all([
```
[the remark this answers is not in the document]
Comment (drewkiimon) [reply packages/extension/src/extension.ts:31]: YOU TELL EM BEN

[the remark this answers is not in the document]
Comment (benawad) [reply packages/extension/src/extension.ts:31]: This looks like it still needs to be reverted back

File: packages/api/src/index.ts [hunk 3]
```diff
@@ -694,7 +692,7 @@ const main = async () => {
       if (["hi", "hello", "hey"].includes((message as string).toLowerCase())) {
         return;
       }
-      User.findOne(userId).then((u) => {
+      await User.findOne(userId).then((u) => {
```
Comment (benawad) [inline packages/api/src/index.ts:695]: I don't want to await this

File: packages/api/src/index.ts [hunk 4]
```diff
@@ -919,11 +917,11 @@ const main = async () => {
       }
 
       if (liked) {
-        User.update(userId, { numLikes: () => '"numLikes" + 1' });
+        await User.update(userId, { numLikes: () => '"numLikes" + 1' });
```
Comment (benawad) [inline packages/api/src/index.ts:920]: I don't want to await this

File: packages/api/src/index.ts [hunk 5]
```diff
@@ -919,11 +917,11 @@ const main = async () => {
...
         wsSend(userId, { type: "new-like" });
       }
 
-      User.update(req.userId, {
+      await User.update(req.userId, {
```
Comment (benawad) [inline packages/api/src/index.ts:924]: I don't want to await this

File: packages/api/src/index.ts [hunk 6]
```diff
@@ -1078,7 +1074,7 @@ const main = async () => {
         ) as any;
         const user = await User.findOne(data.userId);
         // token has been invalidated or user deleted
-        if (!user || user.tokenVersion !== data.tokenVersion) {
+        if (user?.tokenVersion !== data.tokenVersion) {
```
Comment (benawad) [inline packages/api/src/index.ts:1077]: I prefer the other version in case tokenVersion is every undefined

**Review by benawad — CHANGES_REQUESTED** [review]

**Pull request closed without merging**